azavea/raster-foundry
# Add data_footprint to scenes

## Overview

Brief description of what this PR does, and why it is needed.

### Checklist



### Notes

Writing custom serializers because of having too many fields is unpleasant and we should strive to avoid having to do so in the future. There's a lot of magic and a lot of choices between seemingly equivalent things that nonetheless behave differently (`convertTo` vs. `asInstanceOf` vs. `to`, for example, for `JsValue`s).

I [changed](https://github.com/azavea/raster-foundry/compare/feature/add-data-footprint-to-scenes?expand=1#diff-eb70f6a64fbc11cf4e57ebb41735a671R268) the bbox intersection filter from scene to data footprints rather than augmenting the filter list -- it makes more sense to filter based on where we have data. How we handle having one or the other I think is a separate question -- e.g., in the absence of a data footprint, should that just fill from the tile footprint?

## Testing Instructions

* run tests from `sbt`
 * `./scripts/server | grep airflow-worker\|app-server`, wait a bit, observe that scenes still successfully post. This confirms that serialization hasn't ruined everything, even though `data_footprint` isn't yet in the post from airflow

Closes #734 

To track both scene footprints and data footprints, we needed an
additional field on scenes. This change also required overhauling the
serialization of Scene.WithRelated objects, since we exceeded the
Func22/Tuple22/jsonFormat22 limit.
